extensions/datastores/kudu/src/main/java/org/locationtech/geowave/datastore/kudu/operations/KuduRangeRead.java

```diff
@@ -112,24 +121,55 @@ protected KuduRangeRead(
             executeQuery(scanner, results);
           }
         }
+      } else if (dataIds != null) {
+        List<byte[]> dataIdsList = Arrays.stream(dataIds).collect(Collectors.toList());
+        for (final byte[] dataId : dataIdsList) {
+          KuduPredicate partitionPred =
+              KuduPredicate.newComparisonPredicate(
+                  schema.getColumn(KuduField.GW_PARTITION_ID_KEY.getFieldName()),
```

Comment:
For the data index deleter, we directly matched on the data ID field, so we should probably do that to be internally consistent

Reply:
The data ID column is not created when we create the table for data index. There is an 
`if (DataIndexUtils.isDataIndex(tableName)) {
            fields =
                Arrays.stream(fields).filter(KuduField::isDataIndexColumn).toArray(
                    KuduField[]::new);
}`
in createTable in KuduOperations so that only partition key, adapter id and value columns are in the schema.

Reply:
Ah, so we need to fix the deleter then. Maybe we should create a separate row class for the data index, because it's definitely confusing to ignore the data ID column and instead put the data ID in the partition key column.